PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/main/java/com/microsoft/azure/toolkit/intellij/azure/sdk/buildtool/RuleConfig.java

```diff
@@ -27,13 +28,15 @@ class RuleConfig {
      * @param servicesToCheck           List of services to check.
      * @param discouragedIdentifiersMap Map of discouraged identifiers and their suggestions.
      * @param antiPatternMessage        AntiPattern message.
+     * @param antiPatternMessageMap     Map of antiPattern messages.
      */
-    public RuleConfig(List<String> methodsToCheck, List<String> clientsToCheck, List<String> servicesToCheck, Map<String, String> discouragedIdentifiersMap, String antiPatternMessage) {
+    public RuleConfig(List<String> methodsToCheck, List<String> clientsToCheck, List<String> servicesToCheck, Map<String, String> discouragedIdentifiersMap, String antiPatternMessage, Map<String, String> antiPatternMessageMap) {
```

Comment:
the key is the field name from the ruleConfigs.json file ` "anti_pattern_message_map": {
      "no_block": "Calling updateCheckpointAsync() without block() will not do anything, use `block()` or `block` operator with a timeout, or consider using the synchronous version `updateCheckpoint().",
      "with_subscribe": "Instead of `subscribe()`, call `block()` or `block()` with timeout, or use the synchronous version `updateCheckpoint()`"
    }`

so _no_block_ -- the message shown when there's no block method
and _with_subscribe_ -- the message shown when there's a subscribe method used
-- they both suggest using block(), but I thought I'd keep them separate as Anu put in the Code Smells doc since there's a specific warning against using subscribe()

Reply:
I made the refactor and integrated to all rules as well:)